src/utils/choose-template-server-action.ts

```diff
@@ -0,0 +1,19 @@
+'use server';
+import { redirect } from 'next/navigation';
+
+export type FormState = {
+    validationError?: string;
+};
+
+const validTemplateTypes: FormDataEntryValue[] = ['sms', 'email', 'nhs-app', 'letter'];
+
+export const chooseTemplateServerAction = (_: {}, formData: FormData) => {
+    const templateType = formData.get('choose-template-radios');
+
+    if (!templateType || !validTemplateTypes.includes(templateType)) {
+        return {
+            validationError: 'Please select an option'
```

Comment:
Lets establish a pattern for validation where there are component specific errors and general errors.  Although there is only one component in this form at the moment, this may be extended in the future.  I.e. can we setup the validation errors to assign an error to a component.

Reply:
updated